### DISCUSSION TITLE
How do you apply a SpaCy project in practice?

### DISCUSSION THREAD

**Discussion by metalaureate** [explosion/spaCy#11699] [Help: Installation] [answered]
This has got to be such a stupid question but I cannot figure out how to load a package to actually use it once I have run a project workflow. I've read and re-read the documentation and just don't get it, sorry.

I ran the training pipeline on this project: https://github.com/explosion/projects/tree/v3/tutorials/ner_tweets (`spacy project run all`)

And I exported the package (`spacy project run package`). This creates a tar.

But I cannot load the package without errors.

OSError: [E053] Could not read config file from ner_tweets/packages/en_ner_tweets-1.0.0/config.cfg
>>> nlp=spacy.load("./ner_tweets/packages/en_ner_tweets-1.0.0")

How do I actually add the model to the pipeline? 🙃

**Comment by ljvmiranda921** [comment, marked as answer]
Hi @metalaureate , the tar output from the `package` command creates an installable Python package. You need to install it first before loading:

```
pip install path/to/en_ner_tweets-1.0.0.tar.gz
```

Then you can load it into Python by passing the appropriate name in `spacy.load`. To sanity-check if your environment has installed the pipeline, you can run `python -m spacy info`. You can see more from the [docs](https://spacy.io/api/cli#package).

Since you ran the whole training process for `ner_tweets`, you can also load the model directly. Something like this:

```python
import spacy
nlp = spacy.load("path/to/training/model-best")
```

**Reply by metalaureate** [reply]
D'oh
